comparisonModelsExperiment/script/line-level-baseline/decision-trees-line-level.py

```diff
@@ -0,0 +1,173 @@
+    result_dir = f'../../output/{classifierName}-line-level-result/'
+
+    for path in [model_dir, result_dir]:
+        if not os.path.exists(path):    
+            os.makedirs(path) 
+
+    word2vec = get_W2V(dataset_name)
+
+    X_train, y_train = get_XY_train(dataset_name, word2vec)    
+    
+    seed = np.random.seed(0)
+
+    clf = Classifier(n_jobs=24, **params, random_state=seed)
+
+    clf.fit(X_train, y_train)
+
+    test_rels = all_eval_releases[dataset_name][1:]
+
+    for rel in test_rels:
+        test_df = get_df(rel, is_baseline=True)
+
+        test_df = test_df[test_df['file-label']==True]
+        test_df = test_df.drop(['is_comment','is_test_file','is_blank'],axis=1)
+
+        all_df_list = []
+
+        for _, df in tqdm(test_df.groupby('filename')):
+
+            code = df['code_line'].tolist()
+
+            code2d = prepare_code2d(code, to_lowercase)
+
+            code3d = [code2d]
+
+            codevec = get_x_vec(code3d, word2vec)
+            X_test = list(map(lambda v: v[0:150], codevec[0]))
+
+            y_pred = clf.predict(X_test) # true or false
+
+            df['line-score-pred'] = y_pred.astype(int) # 1 or 0, if 1 then the label is correct (ie. l should have no defect and is labeled as no defect), else 0
+
+            all_df_list.append(df)
+
+        all_df = pd.concat(all_df_list)
+
+        all_df.to_csv(result_dir+rel+'-line-lvl-result.csv',index=False)
+
+        print('finished',rel)
+
+
+
+def hyperparameters_tuning(dataset_name, Classifier, params_grid):
+    word2vec = get_W2V(dataset_name)
+    X_train, y_train = get_XY_train(dataset_name, word2vec)    
+    clfSearch = Classifier()
+    searchCV = RandomizedSearchCV(
+        estimator=clfSearch,
+        param_distributions=params_grid,
+        scoring='accuracy',
+        n_iter=25,
+        n_jobs=4,
+        verbose=1
+    )
+    
+    searchCV.fit(X_train, y_train)
+    return searchCV.best_params_
+
+rf_params_grid = {
+    "n_estimators": np.arange(10, 100, 10),
+    "max_depth": [None, 3, 5, 10],
+    "min_samples_split": np.arange(2, 20, 2),
+    "min_samples_leaf": np.arange(1, 20, 2),
+    "max_features": [0.5, 1, "sqrt"],
+    "max_samples": [10000]
+}
+
+# https://xgboost.readthedocs.io/en/stable/parameter.html
+xgb_params_grid = {    
+    "objective": ["binary:logistic", "reg:logistic", "count:poisson"],
+    "tree_method": ["hist"],
+    "eta": [0.1, 0.3, 0.9],
+    'max_depth': [3, 6, 10, 15],
+    'learning_rate': [0.01, 0.1, 0.2, 0.3, 0.4],
+    'subsample': np.arange(0.5, 1.0, 0.1),
+    'colsample_bytree': np.arange(0.5, 1.0, 0.1),
+    'colsample_bylevel': np.arange(0.5, 1.0, 0.1),
+    'n_estimators': np.arange(10, 100, 10),
+}
+
+# https://lightgbm.readthedocs.io/en/latest/Parameters-Tuning.html
+lgbm_params_grid = {
+    "objective": np.array(["regression", "binary", "gamma"]),
+    "n_estimators": np.array([10, 50, 200, 1000]),
+    "learning_rate": np.arange(0.2, 0.9, 0.1),
```

Comment:
with xgb you are using values (0.01, 0.1, 0.2, 0.3, 0.4). Why it changes to  (0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9) here?